config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/util/gateway/ConnectionUtils.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2018. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+package com.ca.apim.gateway.cagatewayconfig.util.gateway;
+
+import com.ca.apim.gateway.cagatewayconfig.tasks.zip.builder.EntityBuilderException;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+public class ConnectionUtils {
+
+    /**
+    * SSLSocketFactory used to retrieve certificates even if they are not in the trust store, to avoid SSL handshake exceptions upon socket connection.
+    * @return the sslSocketFactory that with
+     */
+    public static SSLSocketFactory createAcceptAllSocketFactory() {
+        try {
+            SSLContext sslContext = SSLContext.getInstance("TLS");
+            sslContext.init(null, new TrustManager[]{new X509TrustManager() {
+                private X509Certificate[] accepted;
+
+                @Override
+                public void checkClientTrusted(X509Certificate[] x509Certificates, String s) throws CertificateException {
+                    throw new UnsupportedOperationException();
+                }
+
+                @Override
+                public void checkServerTrusted(X509Certificate[] x509Certificates, String s) throws CertificateException {
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Change this method so it throws exceptions. ([squid:S4424](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS4424))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=68&issues=AWX4rQKo2IDoRcZ5VlPt&open=AWX4rQKo2IDoRcZ5VlPt)

[ISSUE_KEY]: AWX4rQKo2IDoRcZ5VlPt